core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -0,0 +1,212 @@
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * The Load manager which controls load operations.
+ */
+public final class LoadManager {
+  private static final Logger LOG = LoggerFactory.getLogger(LoadManager.class);
+  private final Map<String, LoadInfo>
+          mLoadPathToInfo = Maps.newHashMap();
+  private final JobIdGenerator mJobIdGenerator = new JobIdGenerator();
+  private final Map<Long, Load> mLoads = Maps.newHashMap();
+  private final Scheduler mScheduler = new Scheduler();
+
+  /**
+   * Validate the load information.
+   * @param loadInfo load information
+   * @return boolean value on whether the load is validated for scheduling or not
+   */
+  public boolean validate(LoadInfo loadInfo) {
+    LoadOptions options = loadInfo.getLoadOptions();
+    String path = loadInfo.getPath();
+    if (mLoadPathToInfo.containsKey(path)) {
+      boolean isNewOption = mLoadPathToInfo.get(path).getLoadOptions().equals(options);
+      if (isNewOption) {
+        LoadInfo info = mLoadPathToInfo.get(path);
+        long loadId = info.getId();
+        Load load = mLoads.get(loadId);
+        load.updateOptions(options);
+        /*Only update bandwidth here.*/
+        long newBandWidth = options.getBandwidth();
+        info.getLoadOptions().setBandwidth(newBandWidth);
+      }
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  /**
+   * Schedule a load to run.
+   * @param loadInfo load meta information
+   * @throws ResourceExhaustedException throw ResourceExhaustedException
+   * @throws InterruptedException throw InterruptedException
+   */
+  public void schedule(LoadInfo loadInfo)
+          throws ResourceExhaustedException, InterruptedException {
+    long loadId = loadInfo.getId();
+    Load load = new Load(loadInfo.getId(),
+            loadInfo.getPath(), loadInfo.getLoadOptions());
+    mScheduler.schedule(load);
+    mLoads.put(loadId, load);
+  }
+
+  static class Scheduler {
+    private static final int CAPACITY = 100;
+    private static final long TIMEOUT = 100;
+    private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
+    private final BlockingQueue<Load> mLoadQueue = new LinkedBlockingQueue<>(CAPACITY);
+    private final AtomicInteger mCurrentSize = new AtomicInteger();
+    private final AtomicLong mIdGenerator = new AtomicLong();
+
+    void schedule(Load load)
+            throws ResourceExhaustedException, InterruptedException {
+      if (mCurrentSize.get() == CAPACITY) {
+        throw new ResourceExhaustedException(
```

Comment:
maybe it's too early for schedule part but should we throw runtime exception and let user retry later on?

Reply:
OK, i agree, but we can also expect user to retry if throwing resource exhausted exception?